src/filters.rs

```diff
@@ -1,9 +1,121 @@
-pub fn size<'a>(input : &'a str) -> String{
-    input.len().to_string()
+use std::fmt;
+use std::error::Error;
+
+use value::Value;
+use value::Value::*;
+
+use self::FilterError::*;
+
+#[derive(Debug)]
+pub enum FilterError {
+    InvalidType(String),
+    InvalidArgumentCount(String),
+    InvalidArgument(u16, String) // (position, "expected / given ")
+}
+
+impl fmt::Display for FilterError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            InvalidType(ref e) => write!(f, "Invalid type : {}", e),
+            InvalidArgumentCount(ref e) => write!(f, "Invalid number of arguments : {}", e),
+            InvalidArgument(ref pos, ref e) => 
+                write!(f, "Invalid argument given at position {} : {}", pos, e)
+        }
+    }
+}
+
+impl Error for FilterError {
+    fn description(&self) -> &str {
+        match *self {
+            InvalidType(ref e) => e,
+            InvalidArgumentCount(ref e) => e,
+            InvalidArgument(_, ref e) => e
+        }
+    }
+}
+
+pub type FilterResult = Result<String, FilterError>;
+
+pub fn size(input : &Value, _args: &Vec<Value>) -> FilterResult {
+    match input {
+        &Str(ref s)    => Ok(s.len().to_string()),
+        &Array(ref x)  => Ok(x.len().to_string()),
+        &Object(ref x) => Ok(x.len().to_string()),
```

Comment:
you can abbreviate this to `&Str(ref s) | &Array(ref x) | &Object(ref x) => Ok(s.len().to_string())` if I'm not mistaken